# Upstream-base: 0f5a757947b8778b680d4d1ae7e6df1792a514f2
# Whole original Hall of Fame Cup owner; native numeric context casts and
# temporary diagnostic admission of existing factories62..64/service11.
--- a/src/Game/SH/SHHallOfFameRoom.cpp
+++ b/src/Game/SH/SHHallOfFameRoom.cpp
@@ -1,3 +1,6 @@
+#if defined(MSCHARGED_NATIVE)
+#include <cstdint>
+#endif
 #include "NL/nlDLListContainer.inl"
 #include "Game/SH/SHHallOfFameRoom.h"
 #include "NL/nlFunction.inl"
@@ -223,20 +226,36 @@
     unsigned int nextItem = 0;
     mMode = mode;
     mTrophyTimers[nextItem] = 0.0f;
+#if defined(MSCHARGED_NATIVE)
+    mCupButton.mContext = reinterpret_cast<void*>(static_cast<std::uintptr_t>(nextItem++));
+#else
     mCupButton.mContext = (void*)nextItem++;
+#endif
     for (int i = 0; i < 2; ++i)
     {
         unsigned int item = nextItem++;
         mTrophyTimers[item] = 0.0f;
+#if defined(MSCHARGED_NATIVE)
+        mAwardButtons[i].mContext = reinterpret_cast<void*>(static_cast<std::uintptr_t>(item));
+#else
         mAwardButtons[i].mContext = (void*)item;
+#endif
     }
     for (int i = 0; i < 4; ++i)
     {
         unsigned int item = nextItem++;
         mTrophyTimers[item] = 0.0f;
+#if defined(MSCHARGED_NATIVE)
+        mPlayerCardButtons[i].mContext = reinterpret_cast<void*>(static_cast<std::uintptr_t>(item));
+#else
         mPlayerCardButtons[i].mContext = (void*)item;
-    }
+#endif
+    }
+#if defined(MSCHARGED_NATIVE)
+    mProgressButton.mContext = reinterpret_cast<void*>(static_cast<std::uintptr_t>(nextItem));
+#else
     mProgressButton.mContext = (void*)nextItem;
+#endif
 }
 
 void SHHallOfFameCup::SceneCreated()
@@ -339,7 +358,11 @@
 
 void SHHallOfFameCup::OnItemPointerEnter(unsigned int index, void* context)
 {
+#if defined(MSCHARGED_NATIVE)
+    unsigned int item = static_cast<unsigned int>(reinterpret_cast<std::uintptr_t>(context));
+#else
     unsigned int item = (unsigned int)context;
+#endif
 
     ++mPointerHoverCounts[index];
     if (mTrophyTimers[item] == 0.0f)
@@ -388,7 +411,11 @@
 
 void SHHallOfFameCup::OnItemPointerLeave(unsigned int index, void* context)
 {
+#if defined(MSCHARGED_NATIVE)
+    unsigned int item = static_cast<unsigned int>(reinterpret_cast<std::uintptr_t>(context));
+#else
     unsigned int item = (unsigned int)context;
+#endif
 
     --mPointerHoverCounts[index];
     mTrophyTimers[item] = 0.1f;
@@ -410,7 +437,11 @@
 void SHHallOfFameCup::OnItemPointerPress(unsigned int index, void* context)
 {
     int i;
+#if defined(MSCHARGED_NATIVE)
+    unsigned int item = static_cast<unsigned int>(reinterpret_cast<std::uintptr_t>(context));
+#else
     unsigned int item = (unsigned int)context;
+#endif
 
     mPlayerCardButtons[item].SetPointerState(2, index);
     StopHallOfFameTrophyEffects();
--- a/src/Game/FE/BaseGameSceneManager.cpp
+++ b/src/Game/FE/BaseGameSceneManager.cpp
@@ -499,6 +499,8 @@
     case (SceneList)61:
         newHandler = new (nlMalloc(sizeof(SHHallOfFameSummary), 8, false)) SHHallOfFameSummary(16);
         break;
+#endif
+#if !defined(MSCHARGED_DIAGNOSTIC_FRONTEND_SEQUENCE) || (defined(MSCHARGED_NATIVE) && defined(MSCHARGED_DIAGNOSTIC_FRONTEND_HALL_OF_FAME))
     case (SceneList)62:
         newHandler = new (nlMalloc(sizeof(SHHallOfFameCup), 8, false)) SHHallOfFameCup(0);
         break;
@@ -508,6 +510,8 @@
     case (SceneList)64:
         newHandler = new (nlMalloc(sizeof(SHHallOfFameCup), 8, false)) SHHallOfFameCup(2);
         break;
+#endif
+#if !defined(MSCHARGED_DIAGNOSTIC_FRONTEND_SEQUENCE)
     case (SceneList)65:
         newHandler = new (nlMalloc(sizeof(SHHallOfFamePlayerCard), 8, false)) SHHallOfFamePlayerCard(13);
         break;
--- a/src/Game/Render/FrontEndPresentation_interp.cpp
+++ b/src/Game/Render/FrontEndPresentation_interp.cpp
@@ -119,11 +119,15 @@
         break;
     }
 
+#endif
+#if !defined(MSCHARGED_DIAGNOSTIC_FRONTEND_SEQUENCE) || (defined(MSCHARGED_NATIVE) && defined(MSCHARGED_DIAGNOSTIC_FRONTEND_HALL_OF_FAME))
     case 11:
         ++m_SP;
         m_SP[-1] = IsWidescreen();
         break;
 
+#endif
+#if !defined(MSCHARGED_DIAGNOSTIC_FRONTEND_SEQUENCE)
     case 12:
     {
         const char* name = MSCHARGED_FE_VM_STRING(m_SP[-1]);
